core/src/main/java/tc/oc/pgm/core/CoreModule.java

```diff
@@ -115,6 +129,11 @@ public CoreModule parse(MapFactory context, Logger logger, Document doc)
           serialNumbers.put(owner, serial + 1);
         }
 
+        ImmutableSet<SelfIdentifyingFeatureDefinition> modeList = null;
+        if (!coreEl.getAttributeValue("modes").isEmpty()) {
+          modeList = parseModeSet(Objects.requireNonNull(Node.fromAttr(coreEl, "modes")), context);
+        }
+
         boolean modeChanges = XMLUtils.parseBoolean(coreEl.getAttribute("mode-changes"), false);
```

Comment:
```suggestion
        ImmutableSet<Mode> modes;
        Node modes = Node.fromAttr(coreEl, "modes");
        if (modes != null) {
            if (coreEl.getAttribute("mode-changes") != null)
                throw new InvalidXMLException("Cannot combine modes and mode-changes");
                
            modes = parseModeSet(modes, context); // Specific set of modes
        } else if (XMLUtils.parseBoolean(coreEl.getAttribute("mode-changes"), false)) {
            modes = null; // All modes
        } else {
            modes = ImmutableSet.of(); // No modes
        }
```
Make parsing for `modes` properly support the legacy mode-changes, also make sure to throw an exception if they're combined.

You also need to remove all traces of `modeChanges` in the rest of the codebase, it is no longer a thing since we have a specific set now